src/main.ts

```diff
@@ -471,6 +472,35 @@ export async function createPipeline(config: Config): Promise<string> {
   }
 }
 
+export async function validatePipeline(pipeline: string): Promise<boolean> {
+  if (typeof process.env.VIB_PUBLIC_URL === "undefined") {
+    core.setFailed("VIB_PUBLIC_URL environment variable not found.")
+  }
+
+  const apiToken = await getToken({ timeout: constants.CSP_TIMEOUT })
+  try {
+    core.debug(`Validating pipeline: ${util.inspect(pipeline)}`)
+    const response = await vibClient.post("/v1/pipelines/validate", pipeline, {
+      headers: { Authorization: `Bearer ${apiToken}` },
+    })
+    core.debug(
+      `Got validate pipeline response data : ${JSON.stringify(
+        response.data
+      )}, headers: ${util.inspect(response.headers)}`
+    )
+
+    if (response.status === 200) {
+      core.info(ansi.bold("The pipeline has been successfully validated."))
+    } else if (response.status === 400) {
+      core.setFailed("The pipeline given is not correct.")
```

Comment:
In addition to fail the GitHub Action you need to display the validation error to the user.